### ISSUE TITLE
SPInstall: Does not detect a paused Installation

### ISSUE THREAD

**Issue by ChristophHannappel** [PowerShell/SharePointDsc#1098]
#### Details of the scenario you tried and the problem that is occurring
When Installing SharePoint 2019 the Ressource finished very quickly but the Product was not installed completly. I also couldn't see anything inside the Programs and Feature Control Panel.
Some files where allready copied onto the destination folders.
When i ran the setup manually the Setup told me that a current installation of the product is paused and if i want to remove its progress. Than the installer went through and i removed SharePoint via Programs and Features. After that the Ressource worked successfully.

I think the issue itself hat its source in a server reboot while dsc was installing SharePoint and left it in a paused state.
#### Verbose logs showing the problem
VERBOSE: [ServerName]: LCM:  [ Start  Resource ]  [[SPInstall]SP-Install]
VERBOSE: [ServerName]: LCM:  [ Start  Test     ]  [[SPInstall]SP-Install]
VERBOSE: [ServerName]:                            [[SPInstall]SP-Install] Testing install status of SharePoint
VERBOSE: [ServerName]:                            [[SPInstall]SP-Install] Getting install status of SharePoint
VERBOSE: [ServerName]:                            [[SPInstall]SP-Install] Check if Binary folder exists
VERBOSE: [ServerName]:                            [[SPInstall]SP-Install] Checking file status of \\FileServer\FileShare\Install\Microsoft\SharePoint Serve
r 2019\Setup\setup.exe
VERBOSE: [ServerName]:                            [[SPInstall]SP-Install] Checking status now
VERBOSE: [ServerName]:                            [[SPInstall]SP-Install] File not blocked, continuing.
VERBOSE: [ServerName]:                            [[SPInstall]SP-Install] Current Values: BinaryDir=\\\FileServer\FileShare\Install\Microsoft\SharePoint Ser
ver 2019\Setup; DataPath=F:\Program Files\Microsoft Office Servers; Ensure=Present; InstallPath=E:\Program Files\Microsoft Office Servers; IsSingleInstance=Yes; Produ
ctKey=7G7R6-N6QJC-JFPJX-CK8WX-66QW4
VERBOSE: [ServerName]:                            [[SPInstall]SP-Install] Target Values: BinaryDir=\\\FileServer\\FileShare\Install\Microsoft\SharePoint Serv
er 2019\Setup; DataPath=F:\Program Files\Microsoft Office Servers; Ensure=Present; InstallPath=E:\Program Files\Microsoft Office Servers; IsSingleInstance=Yes; Produc
tKey=<I'm a product key>; Verbose=True
VERBOSE: [ServerName]: LCM:  [ End    Test     ]  [[SPInstall]SP-Install]  in 0.8260 seconds.
VERBOSE: [ServerName]: LCM:  [ Skip   Set      ]  [[SPInstall]SP-Install]
VERBOSE: [ServerName]: LCM:  [ End    Resource ]  [[SPInstall]SP-Install]

#### Suggested solution to the issue
Try to check if the installation is paused and throw an error. 
#### The DSC configuration that is used to reproduce the issue (as detailed as possible)
```powershell
SPInstall SP-Install
            {
                Ensure = 'Present'
                DependsOn = '[SPInstallPrereqs]SP-Prerequisits','[Disk]Disk-SPBin','[Disk]Disk-SPData'
                IsSingleInstance = 'Yes'
                BinaryDir = $ConfigurationData.NonNodeData.SPSetupPath
                ProductKey = $ConfigurationData.NonNodeData.SPProductKey
                InstallPath = '<Drive>:\Program Files\Microsoft Office Servers'
                DataPath = '<Drive>:\Program Files\Microsoft Office Servers\16.0\Data'
                PsDscRunAsCredential = $SPDSCAccount

            }
```

#### The operating system the target node is running
OsName               : Microsoft Windows Server 2016 Datacenter
OsOperatingSystemSKU : DatacenterServerEdition
OsArchitecture       : 64-bit
WindowsBuildLabEx    : 14393.3024.amd64fre.rs1_release.190530-2002
OsLanguage           : en-US
OsMuiLanguages       : {en-US}

#### Version of SharePoint that is used (e.g. SharePoint 2016)
SharePoint 2019

#### Version and build of PowerShell the target node is running
Name                           Value                                                                                                                                 
----                           -----                                                                                                                                 
PSVersion                      5.1.14393.2969                                                                                                                        
PSEdition                      Desktop                                                                                                                               
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                                                               
BuildVersion                   10.0.14393.2969                                                                                                                       
CLRVersion                     4.0.30319.42000                                                                                                                       
WSManStackVersion              3.0                                                                                                                                   
PSRemotingProtocolVersion      2.3                                                                                                                                   
SerializationVersion           1.1.0.1    

#### Version of the DSC module that was used ('dev' if using current dev branch)
3.5.0.0

**Comment by ykuijs** [comment]
Hi @ChristophHannappel, this is a good one. I haven't seen a paused installation before.

SPInstall uses the registery to determine if SharePoint is installed or not. With the release of SP2019, several components were renamed. For example the Foundation Core component now also includes "Microsoft SharePoint Server" in the name. If the installation completes installing this component and then fails, the registry still returns results causing SPInstall to think SP2019 has been installed.

I just investigated #1099 and found a solution for that issue which I think will also fix this issue.

**Comment by ykuijs** [comment]
This issue is now fixed in dev and will be released with v3.7
